Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+HTTP1StateMachine.swift

```diff
@@ -219,6 +223,17 @@ extension HTTPConnectionPool {
 
             switch self.lifecycleState {
             case .running:
+                guard self.retryConnectionEstablishment else {
+                    guard let (index, _) = self.connections.failConnection(connectionID) else {
+                        preconditionFailure("Failed to create a connection that is unknown to us?")
```

Comment:
I often write the precondition messages like this... But I actually consider this bad style. This might be better:
```suggestion
                        preconditionFailure("A connection attempt failed, that the state machine knows nothing about. Somewhere state was lost.")
```